### PR TITLE
Adding additional wait for android device to come up.

### PR DESCRIPTION
**Pull request by yusuf-goog**
Bug:flutter/flutter#135035

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: cipd_packages/device_doctor/lib/src/android_device.dart [hunk 1]
```diff
@@ -453,14 +453,14 @@ class AndroidDevice implements Device {
     processManager ??= LocalProcessManager();
     final int timeoutSecs = 60;
     print('Device recovery: deleting package caches...');
-    await eval('adb', <String>['wait-for-device'], canFail: false, processManager: processManager)
+    await eval('adb', <String>['wait-for-device', 'shell', 'while [[ -z \$(getprop sys.boot_completed) ]]; do sleep 1; done;'], canFail: false, processManager: processManager)
```
Comment (ricardoamador) [inline cipd_packages/device_doctor/lib/src/android_device.dart:456]: FYI: this is what I was referring to when I tried this through similar code. Are you able to run this command through the ProcessManager and have it succeed without error?

Comment (ricardoamador) [reply cipd_packages/device_doctor/lib/src/android_device.dart]: See this: https://discord.com/channels/608014603317936148/846507907876257822/1156718853921964052

File: cipd_packages/device_doctor/lib/src/android_device.dart [hunk 2]
```diff
@@ -453,14 +453,20 @@ class AndroidDevice implements Device {
     processManager ??= LocalProcessManager();
     final int timeoutSecs = 60;
     print('Device recovery: deleting package caches...');
-    await eval('adb', <String>['wait-for-device'], canFail: false, processManager: processManager)
+    await eval('adb',
+            <String>['wait-for-device', 'shell', 'while [[ -z \$(getprop sys.boot_completed) ]]; do sleep 1; done;'],
+            canFail: false, processManager: processManager)
         .timeout(Duration(seconds: timeoutSecs));
     await deletePackageCache();
```
Comment (ricardoamador) [inline cipd_packages/device_doctor/lib/src/android_device.dart:460]: Are these calls restarting the device?

Comment (yusuf-goog) [reply cipd_packages/device_doctor/lib/src/android_device.dart:460]: Aside from the reboot, it should be. I think the issue is the adb server and whats calling this script vs other things on the bot.

File: cipd_packages/device_doctor/lib/src/android_device.dart (hunk 1 above)
Comment (yusuf-goog) [reply cipd_packages/device_doctor/lib/src/android_device.dart]: What is the easier way you found?

Comment (ricardoamador) [reply cipd_packages/device_doctor/lib/src/android_device.dart]: I just did not include the loop as I could not get it to run.

Comment (ricardoamador) [reply cipd_packages/device_doctor/lib/src/android_device.dart]: The wait-for-device should at least be able to detect the attached device.

File: cipd_packages/device_doctor/lib/src/android_device.dart (hunk 2 above)
Comment (ricardoamador) [reply cipd_packages/device_doctor/lib/src/android_device.dart:462]: The delete calls don't restart the device. I guess the other question in regards to those is how does it wait for the package to be deleted and how long does it take to delete a package?

Comment (yusuf-goog) [reply cipd_packages/device_doctor/lib/src/android_device.dart:462]: This is hapepning before the package delete call itself

Comment (ricardoamador) [reply cipd_packages/device_doctor/lib/src/android_device.dart:462]: Further up in the call stack it attempts to list devices and if it cannot it kills the adb server for some reason.
https://github.com/flutter/cocoon/blob/main/cipd_packages/device_doctor/lib/src/android_device.dart#L58

**Comment by ricardoamador** [comment]
@yusuf-goog are you still planning on landing this?

**Comment by keyonghan** [comment]
Closing as not needed.

**Pull request closed without merging**